docs/blueprints/EP038.rst

```diff
@@ -0,0 +1,151 @@
+:EP: 38
+:Title: Broadcasting events
+:Authors:
+- Luis;
+- Italo;
+- Vinicius;
+- Jeronimo Bezerra jbezerra@fiu.edu;
+:Issued Date: to be defined
+:Status: Pending
+:Type: Standards Track
+
+EP03X - Broadcasting events
+
+########
+Abstract
+########
+
+This blueprint details how Kytos should export/broadcast its events
+to external applications in a safe and scalable way.
+
+##########
+Motivation
+##########
+
+With Kytos running in production, we learned that Kytos events can be
+leveraged for multiple purposes, from capacity planning to topology evaluation,
+from troubleshooting to accounting. However, creating Kytos napps to only monitor the environment should be avoided to minimize possible impacts on running the kytos instance. Moreover, this napp has the potential to expedite the prototyping of new applications and frameworks that need an understanding of the
+network topology and control plane, such as the AtlanticWave-SDX project.
+
+##########
+Requirements
+##########
+
+This napp should not be named after or delegate part of its functions to the message broker system
+This napp has to be an event broadcaster only. No input via message_bus should be supported/allowed.
```

Comment:
The "topology_event" Napp will communicate with the topic exchange within the "mq_producer" Napp. This topic exchange operates similarly to a direct exchange but introduces a more adaptable routing mechanism based on routing patterns. Unlike direct exchanges, which rely on fixed routing keys, topic exchanges utilize wildcards for message routing, enhancing flexibility.

Here's a breakdown of its operation:

Instead of depending on specific routing keys, topic exchanges route messages by comparing a message's routing key with predefined patterns.
This comparison determines the routing of messages to one or more queues based on their correspondence with the specified patterns.
The routing key consists of a series of words separated by periods (".").
In summary, the topic exchange facilitates nuanced and dynamic message routing through wildcard patterns, offering increased flexibility and versatility in distributing messages within the messaging system.